google/cloud/aiplatform/compat/services/__init__.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from google.cloud.aiplatform_v1beta1.services.dataset_service import (
+  client as dataset_service_client
+)
+from google.cloud.aiplatform_v1beta1.services.endpoint_service import (
+  client as endpoint_service_client,
+)
+from google.cloud.aiplatform_v1beta1.services.job_service import (
+  client as job_service_client,
+)
+from google.cloud.aiplatform_v1beta1.services.model_service import (
+  client as model_service_client,
+)
+from google.cloud.aiplatform_v1beta1.services.pipeline_service import (
+  client as pipeline_service_client,
+)
+from google.cloud.aiplatform_v1beta1.services.prediction_service import (
+  client as prediction_service_client,
+)
+
+__all__ = (
+    dataset_service_client,
+    endpoint_service_client,
+    job_service_client,
+    model_service_client,
+    pipeline_service_client,
+    prediction_service_client,
```

Comment:
I like this idea. It seems like we should abstract away the client version handling away from the resource noun implementation as there's only one client. We have a few options there:

1. Is include v1beta1 and v1 clients in there which may defeat the purpose of this module
2. The other is wrap both clients and forward the request to the correct client. I created this quick POC implementation here: https://github.com/sasha-gitg/python-aiplatform/commit/93ab9766e11eac82e36498fad9923fc64f2d5baa

That way each resource noun has a single client without the need to manage versions. ie:

```python
client = prediction_service_client()
client.predict(...)  # goes to v1
client.explain(...)  # goes to v1beta1
```